enhancements/testing/improved-platform-tests.md

```diff
@@ -0,0 +1,456 @@
+   or [kubernetes/rules.go](https://github.com/openshift/kubernetes/blob/8bd5514be14de7fb3105962c8fc206ef5e921a0b/openshift-hack/e2e/annotate/rules.go),
+   in combination with [the annotation mechanism](https://github.com/openshift/kubernetes/blob/8bd5514be14de7fb3105962c8fc206ef5e921a0b/openshift-hack/e2e/annotate/annotate.go).
+3. Help managing the use of disconnected images (i.e. ensure they are only referencing
+   images that are mirrored, at least for conformance tests). See [images used by e2e tests](https://github.com/openshift/origin/blob/cc42164781a728b804d0bb07f09cf878ec8f7807/test/extended/util/image/README.md).
+4. Provide an image annotation or similar mechanism informing where the test
+   binary is embedded in the release payload.
+
+The proposed extensions to `openshift-tests` will require the following changes:
+1. Add extraction logic responsible for retrieving test binaries from the
+   release payload, based on the annotations provided by the authors of these
+   images.
+2. Add logic responsible for iterating over all test binaries invoking the `list`
+   subcommand, and reading the list of tests provided by the binary. Every tests
+   should have an associated binary name, which will be used during invocation.
+3. Expand the current filtering, shuffling, and randomization logic to also take
+   into account the new tests coming from external binaries.
+4. Add logic responsible for iterating over all tests, invoking the `run`
+   subcommand using the associated test's binary and reading its results.
+5. Provide a fallback path allowing to use the vendored tests on demand (`OPENSHIFT_SKIP_EXTERNAL_TESTS`
+   environment variable, for example).
+
+### Risks and Mitigations
+
+- **The current policy when the [origin](https://github.com/openshift/origin/) pull
+  request bringing the updated k8s version breaks the CI is to revert PR. This
+  policy will be harder to execute with this change because reverting the o/k
+  rebase PR can be problematic and was never tried before.**
+
+  There should be fewer reasons to do any reverting, because with this new mechanism
+  we should get the appropriate signal during extensive payload testing required
+  to land every k8s bump PR.
+
+- **Some of the work that has been done to build the partner certification tool will
+  need to be updated to pull additional images with separate test binaries.**
+
+  The `openshift-tests` binary will contain the necessary logic to ensure that test
+  invocation is working as before.
+
+- **Not available release image can prevent from running the tests.**
+
+  The proposed mechanism will contain a fallback path, which will ensure that in
+  cases where the release payload is not reachable we will be able to run all the
+  tests bundled within the `openshift-tests` binary. This includes the basic
+  OpenShift and Kubernetes conformance tests.
```

Comment:
Meaning the openshift/origin test suite will always have the latest k8s e2e's present?

Reply:
For now - yes, see [Fabio's comment](https://github.com/openshift/enhancements/pull/1291#discussion_r1246863838), and the other case is Microshift. There's a long-term goal to eventually get rid of that, but we're not there yet.